app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx

```diff
@@ -317,11 +330,12 @@ function SubSectorPage({
             </MotionTabList>
 
             <TabPanels>
-              {isLoading ? (
+              {isLoading || !refNumber ? (
```

Comment:
This means that if the refNumber is missing, it keeps showing the loading spinner indefinitely. Better behavior might be to show a warning icon and/ or message (can be coordinated with the design team, for now a warning icon is sufficient).